vm/actor/tests/common/mod.rs

```diff
@@ -128,6 +130,26 @@ impl<'a, BS: BlockStore> MockRuntime<'a, BS> {
     pub fn expect_validate_caller_any(&self) {
         self.expect_validate_caller_any.set(true);
     }
+
+    #[allow(dead_code)]
+    pub fn expect_call_fail(
+        &mut self,
+        to_code: &Cid,
+        method_num: MethodNum,
+        params: &Serialized,
+        expected_code: ExitCode,
+    ) {
+        // will record previous state, execute call. Will expect a certain exit code. Will throw assertion fail if exit code does not match
+        //  if exit code matches will revert state
+        let prev_state = self.state.clone();
+
+        let call_result = self.call(to_code, method_num, params);
+
+        assert_eq!(expected_code, call_result.unwrap_err().exit_code());
+
+        self.state = prev_state;
+    }
```

Comment:
Okay so my question for this is, why not just add a revert if error at the end of our mock rt call function? Seems weird it would be setup to have to manually revert for expected failures and not for a call. @ec2 maybe you have an opinion on this?

Reply:
I think whichever is fine. Its just doing this could make it easier to port tests since theres a lot less thinking since this is how they do it specs actor

Reply:
In a way, yes, but if you are referring to this being easier, an example of how it isn't is in the case of the init actor: it can't use this function because there is other logic that is needed before the call. 

This function is setup for a very specific use case so probably won't be applicable in most other tests.

Reply:
The way I suggested, you get that for free and don't have to do that logic in different places. Just sharing thought process before I forget, I don't care about this, it's just tests but may make life easier

Reply:
So do you think its better to just change the function signature for call? Have a parameter called expected_code to help decide whether or not to revert states?